### DISCUSSION TITLE
reshade built-in

### DISCUSSION THREAD

**Discussion by Rosalie241** [gonetz/GLideN64#2443] [General]
hey @gonetz,

would you be opposed to a patch which includes some reshade code to make reshade shaders work with GLideN64 without requiring reshade itself?

The benefits of such a change would include linux support for reshade shaders in GLideN64 for example, which'd be cool in my opinion!

I'm mainly asking this because I don't want to work on something for a while and then get my patch rejected.

**Comment by gonetz** [comment]
Hi!
I have no plans to work on new post-processing shaders but I don't mind if you or somebody else implement that task.
It should be a separate library with plain C API, like GLideNHQ, so it can be easily excluded from the build.
"some reshade code" - is Reshade team agree with it?
Reshade shaders have many settings. How do you plan to implement them? The same way as it is done in Rashade, that is without dialog window?

**Reply by Rosalie241** [reply]
>  It should be a separate library with plain C API, like GLideNHQ, so it can be easily excluded from the build.

that's fine.
>  "some reshade code" - is Reshade team agree with it?

reshade is opensource: https://github.com/crosire/reshade and if you look at the README, it mentions the following:
>> The ReShade FX shader compiler contained in this repository is standalone, so can be integrated into other projects as well. 
Simply add all `source/effect_*.*` files to your project and use it similar to the fxc example.


>  Reshade shaders have many settings. How do you plan to implement them? The same way as it is done in Rashade, that is without dialog window?

I'm considering modifying GLideNUI to display those settings.

**Reply by gonetz** [reply]
> I'm considering modifying GLideNUI to display those settings.

It can be the hardest part of that task then. Reshade has lots of shaders with plenty of settings. Do you plan to write some dynamic generation of GUI controls for all these settings? Is it possible to re-use Reshade's way of changing settings?

**Reply by Rosalie241** [reply]
> It can be the hardest part of that task then. Reshade has lots of shaders with plenty of settings. Do you plan to write some dynamic generation of GUI controls for all these settings?

yeah dynamic generation of GUI controls was my idea, maybe I'll store the actual settings in a different config file though as to not clutter `GLideN64.ini`, maybe i.e `shaders.ini` to store effects filenames, settings etc

>  Is it possible to re-use Reshade's way of changing settings?

reshade uses imgui and I think introducing another dependency would be bad (let alone writing imgui wrappers to actually be able to get a reshade-like interface would be quite a big task)

**Reply by gonetz** [reply]
> reshade uses imgui and I think introducing another dependency would be bad

Hard dependency for the main module (GLideN64) would be bad indeed.
If we talk about optional sub-module, we can consider various scenarios.
GLideNUI depends on Qt. Qt is too large to be anyhow included into GLideN64 repository. If you need to build GLideNUI, you should not only clone GLideN64 repository, but also download precompiled Qt libraries from some external sources.
I think, you may use ImGui the same way as we already use Qt. As I understand, it can be statically linked, so we still will have one dll file. File size will increase, but as soon as it is optional, it is acceptable. User may take lighter build if that functionality is not necessary.
I even think, that we can try to implement GLideN64 GUI with ImGui, at least partially. It would have such benefits as instant apply of changed setting or ability to change settings in full screen mode. If you can do this in the new module, its importance will increase greatly.
